### DISCUSSION TITLE
Failed to create PNG image

### DISCUSSION THREAD

**Discussion by jchan192** [GLVis/glvis#214] [Q&A] [answered]
Dear Glvis Community,

I have the following mesh and gridfunction files produced by MFEM, 
which can be visualized successfully by GLVIS,
[mesh_and_gf.tar.gz](https://github.com/GLVis/glvis/files/7761220/mesh_and_gf.tar.gz)
 
However, I always get a black png when I try to screenshot it with a script like this
```
window 0 0 300 300

solution A.mesh A.gf screenshot result.png
```
I wonder if anyone knows how to resolve the issue here.
Thank you so much.

**Comment by v-dobrev** [comment, marked as answer]
For me the script works fine on macOS.

One thing to remember is that when the GLVis window opens, it stops reading the script after the "solution mesh-file sol-file" command, so it will not automatically execute the following commands. To do that, one needs to press the space key in the GLVis window. I suspect you are already doing that, otherwise you will not get the file "result.png".

One thing you can try is to test if screenshots work for you without using a script. Can you try running the command
```console
glvis -m A.mesh -g A.gf
```
and then take a screenshot by pressing the `S` key. Is the generated file (`GLVis_s01.png`) a black png as well? If that is the case, can you give us more details about how you built GLVis and on what platform, so we can try to reproduce and debug the issue?

**Reply by tzanio** [reply]
This looks to be a periodic AMR mesh. It takes a while on Mac laptop, but I can confirm that the following works:

```
glvis -m A.mesh -g A.gf -k "imMaacS"
```

producing this image:

<img width="400" alt="GLVis_s01" src="https://user-images.githubusercontent.com/1278247/147135446-9a1ee848-f910-4e41-9064-c00a8489dda1.png">

**Comment by jchan192** [comment]
Thanks for confiming it for me. 
After I read your message, I tried to run the GLVis script in my another (windows) laptop and it works! 
So somehow my Linux station failed to screenshot PNGs with GLVis script, but at this moment I will use my laptop
isntead then. 

I hope you don't mind me asking another related question here.
I could press `i` and Toggle cutting (clipping) plane just like @tzanio  in a GLVis window,
but I wonder what is the command line to do it in the GLVis script?

I could not find a manuel that describes commands for GLVis script.
Thanks

**Comment by v-dobrev** [comment]
You can use the `keys` script command followed by a sequence of keys. It is implemented here: https://github.com/GLVis/glvis/blob/92595f9c5ee58e729bc2368d452039eb80eeee56/glvis.cpp#L650-L657  You can see the other script commands in the same function -- I don't think we've documented them anywhere yet.
